sympy/diffgeom/diffgeom.py

```diff
@@ -321,85 +514,133 @@ def coord_tuple_transform_to(self, to_sys, coords):
         return BaseScalarField(self, coord_index)
+    coord_function = base_scalar
 
-    def coord_functions(self):
+    def base_scalars(self):
         """Returns a list of all coordinate functions.
-
-        For more details see the ``coord_function`` method of this class."""
-        return [self.coord_function(i) for i in range(self.dim)]
+        For more details see the ``base_scalar`` method of this class."""
+        return [self.base_scalar(i) for i in range(self.dim)]
+    coord_functions = base_scalars
 
     def base_vector(self, coord_index):
         """Return a basis vector field.
-
         The basis vector field for this coordinate system. It is also an
-        operator on scalar fields.
-
-        See the docstring of ``CoordSystem`` for examples."""
+        operator on scalar fields."""
         return BaseVectorField(self, coord_index)
 
     def base_vectors(self):
         """Returns a list of all base vectors.
-
         For more details see the ``base_vector`` method of this class."""
         return [self.base_vector(i) for i in range(self.dim)]
 
     def base_oneform(self, coord_index):
         """Return a basis 1-form field.
-
         The basis one-form field for this coordinate system. It is also an
-        operator on vector fields.
-
-        See the docstring of ``CoordSystem`` for examples."""
+        operator on vector fields."""
         return Differential(self.coord_function(coord_index))
 
     def base_oneforms(self):
         """Returns a list of all base oneforms.
-
         For more details see the ``base_oneform`` method of this class."""
         return [self.base_oneform(i) for i in range(self.dim)]
 
-    ##########################################################################
-    # Points.
-    ##########################################################################
+class CoordinateSymbol(Symbol):
+    """A symbol which denotes an abstract value of i-th coordinate of
+    the coordinate system with given context.
 
-    def point(self, coords):
-        """Create a ``Point`` with coordinates given in this coord system.
+    Explanation
+    ===========
 
-        See the docstring of ``CoordSystem`` for examples."""
-        return Point(self, coords)
+    Each coordinates in coordinate system are represented by unique symbol,
+    such as x, y, z in Cartesian coordinate system.
 
-    def point_to_coords(self, point):
-        """Calculate the coordinates of a point in this coord system.
+    You may not construct this class directly. Instead, use `symbols` method
+    of CoordSystem.
 
-        See the docstring of ``CoordSystem`` for examples."""
-        return point.coords(self)
+    Parameters
+    ==========
 
-    ##########################################################################
-    # Printing.
-    ##########################################################################
+    coord_sys : CoordSystem
+
+    index : integer
+
+    Examples
+    ========
```

Comment:
I'm not sure it documents about the usage of creating coordinate systems without symbols
```
m = Manifold('M', 2)
p = Patch('P', m)
C = CoordSystem('C', p)
C.symbols
```

And now I get some extraneous warning when I create coordinate systems without symbols

```
Class signature 'names' of CoordSystem has been deprecated since SymPy
1.7. Use class signature 'symbols' instead. See
https://github.com/sympy/sympy/issues/19321 for more info.

  deprecated_since_version="1.7"
```
is it intended feature to deprecate such?

Reply:
This wasn't my intention. I fixed it. Thanks!